src/config/config_type.rs

```diff
@@ -431,18 +431,24 @@ macro_rules! create_config {
                     heuristic_value: usize,
                     config_key: &str,
                 | -> usize {
-                    if !was_set {
-                        return heuristic_value;
-                    }
-                    if override_value > max_width {
+                    let value = if !was_set {
+                        heuristic_value
+                    } else {
+                        override_value
+                    };
+
+                    let value = value.min(max_width);
+
+                    if value > max_width {
```

Comment:
Can we ever actually issue the warning if we clamp the width value before evaluating `value > max_width`?

Instead of making changes to `set_width_heuristics ` I think it would be nice to add a new API that sets the `max_width` and also clamps the other width settings.

Reply:
Are you saying you want to use your proposed new API to change in the macro code

```rust
let new_width = config.max_width() - body_indent.width();
config.set().max_width(new_width);
```

to something else?  If so, can you give me a clearer idea of what that API would be?  Maybe:

```rust
config.set_clamp().max_width(new_width);
```

Alternatively, with no new API, the clamp could be made unconditional in the `ConfigSetter`, which would address your concern about modifying `set_width_heuristics`.

Please let me know what you think.

Reply:
I went ahead and created this API.  Let's see what you think.